src/util/amount.rs

```diff
@@ -160,7 +160,7 @@ fn parse_signed_to_satoshi(
             // there are no decimals and the last digits are zeroes as
             // many as the difference in precision.
             let last_n = precision_diff.abs() as usize;
-            if s.contains(".") || s.chars().rev().take(last_n).any(|d| d != '0') {
+            if s.contains(".") || last_n >= s.len() || s.chars().rev().take(last_n).any(|d| d != '0') {
```

Comment:
I would make this `bool is_too_precise(&str)`. Then it is self-explanatory, and you can add couple of doctests and stuff.